### PR TITLE
Ensure that Chrome closes on error/exception

### PR DESCRIPTION
**Pull request by ramlmn**
Fixes issue #1002

### CODE REVIEW

File: lighthouse-cli/bin.ts [hunk 1]
```diff
@@ -215,7 +215,11 @@ function launchChromeAndRun(addresses: Array<string>,
         return launcher.run();
       })
       .then(() => lighthouseRun(addresses, config, flags))
-      .then(() => launcher.kill());
+      .then(() => launcher.kill())
+      .catch((error) => {
```
Comment (ebidel) [inline lighthouse-cli/bin.ts]: `(error) =>` -> `error =>`

File: lighthouse-cli/bin.ts [hunk 2]
```diff
@@ -215,7 +215,11 @@ function launchChromeAndRun(addresses: Array<string>,
...
+        console.error(error);
```
Comment (ebidel) [inline lighthouse-cli/bin.ts]: The stacktrace isnt very user friendly. Let's just log the error using the log method:

`log.error('Lighthouse CLI', error.message);`

**Comment by brendankenny** [comment]
I think I prefer the second option, as that keeps the `reject` paths together, which will make the eventual refactor slightly simpler to reason through.

**Comment by ramlmn** [comment]
@brendankenny The second option seems fine to me too. Maybe we can do something like
``` js
.catch(error => {
  log.error('Lighthouse CLI', error.message);
  launcher.kill()
    .then(_ => {
      process.emit('SIGINT');
    });
});
```
or just throwing the error again.

Whatever you like...

**Comment by ramlmn** [comment]
Can someone take a look at this PR again...!?

**Comment by brendankenny** [comment]
@ramlmn sorry, I didn't realize you were asking a question. I think the second option (replacing [this line in `run()`](https://github.com/GoogleChrome/lighthouse/blob/558a26f8661565caebcbba085b26bf028a89c283/lighthouse-cli/bin.ts#L294) is better rather than added this as in the current form of the PR.

**Comment by paulirish** [comment]
this needs to happen inside of 

```js
       return Promise
         .race([launchChromeAndRun(urls, config, cliFlags), isSigint])
         .catch(maybeSigint => {
```

because right now you're stealing exceptions from the error handling that already exists at the bottom of the file.

**Comment by ramlmn** [comment]
>because right now you're stealing exceptions from the error handling that already exists at the bottom of the file.

@paulirish you are right but the problem is that `cleanup.doCleanup` is only called when there is a `_SIGINT` exception *only*. We actually try to emit `'SIGINT'` so that `doCleanup` gets called, but that seems bad. So, just we can do something like:
``` js
return Promise
  .race([launchChromeAndRun(urls, config, cliFlags), isSigint])
  .catch(maybeSigint => {
  if (maybeSigint === _SIGINT || maybeSigint instanceof Error) {
    if (maybeSigint instanceof Error) {
      log.error("Lighthouse", maybeSigint.message);
    }
    return cleanup
      .doCleanup()
      .catch(err => {
        console.error(err);
        console.error(err.stack);
      }).then(() => process.exit(_ERROR_EXIT_CODE));
  }
  return handleError(maybeSigint);
});
```
Because `maybeSigint` can also be an `Error` of `launcher` from `launchChromeAndRun` or any other error, in that case we can just cleanup and exit.

I tried to emit `SIGINT` because I didn't realize we had access to `launcher.kill()` from inside `cleanup`.

What do you all think about this? /cc @ebidel @brendankenny

**Comment by ramlmn** [comment]
Can you all review this PR? @ebidel @paulirish @brendankenny

**Comment by brendankenny** [comment]
hi @ramlmn, this was fixed in the course of the refactor in #1141 so that in all paths of running the CLI an error will close Chrome if launched by Lighthouse. I'm going to close as fixed. Please feel free to reopen or file a new issue if there's still an issue for you.

**Pull request closed without merging**